client/components/common/FileUpload.vue

```diff
@@ -45,13 +45,20 @@ export default {
     id: { type: String, default: () => uniqueId('file_') },
     fileName: { type: String, default: '' },
     fileKey: { type: String, default: '' },
-    validate: { type: Object, default: () => ({ rules: { ext: [] } }) },
+    validate: { type: Object, default: () => ({ ext: [] }) },
     label: { type: String, default: 'Choose a file' },
     sm: { type: Boolean, default: false }
   },
   data() {
     return { uploading: false };
   },
+  computed: {
+    fileFilter() {
+      const { validate: { ext, rules } } = this;
+      if (Array.isArray(ext)) return ext.join(',');
+      if (rules && Array.isArray(rules.ext)) return rules.ext.join(',');
```

Comment:
Actually we can avoid re-implementing `vee-validate`'s internals by leveraging `Field` properties: https://baianat.github.io/vee-validate/api/field.html#api
```js
fileFilter() {
  const field = this.$validator.vFields({ name: this.id });
  if (!field || !Array.isArray(field.rules.ext)) return '';
  return field.rules.ext.join(',');
}
```